opaque_keys/__init__.py

```diff
@@ -327,10 +371,11 @@ def replace(self, **kwargs):
             return self
 
         existing_values.update(kwargs)
+
         return type(self)(**existing_values)
 
     def __setattr__(self, name, value):
-        if getattr(self, '_initialized', False):
+        if name != '_unicode' and name != '_hash' and name != '_cached_key' and getattr(self, '_initialized', False):
```

Comment:
Good point. I'll try it out. Though honestly, the `__setattr__` isn't called very often relative to the others, since there are three cachable fields, and < 5K keys generated now, even on large courses.